analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd

```diff
@@ -0,0 +1,229 @@
+---
+title: "Integrate molecular subtyping results"
+output: 
+  html_notebook:
+    toc: true
+    toc_float: true
+author: Krutika Gaonkar for D3b
+date: 2020
+params:
+   is_ci: FALSE
+---
+
+The purpose of this notebook is to integrate molecular subtyping results from 
+[molecular-subtyping-pathology](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-pathology) with `pbta-histologies-base.tsv`.
+
+Here we are using v18 `pbta-histologies-base.tsv` which has integrated_diagnosis,Notes and molecular_subtype as NA. Through all the following molecular subtyping modules:
+
+- molecular-subtyping-MB
+- molecular-subtyping-CRANIO
+- molecular-subtyping-EPN
+- molecular-subtyping-embryonal
+- molecular-subtyping-EWS
+- molecular-subtyping-neurocytoma
+- molecular-subtyping-HGG
+- molecular-subtyping-LGAT
+- molecular-subtyping-pathology
+
+we gathered and updated molecular-subtype/integrated_diagnosis/broad_histology/short_histology. 
+
+In this notebook we will add the molecular subtyping done as part as OpenPBTA to the base histology file to create the `pbta-histologies.tsv` for the same release. 
+
+For samples that where pathology_diagnosis is "Other" we also update the file broad_histology and short_histology from a manual review of WHO terms.
+
+## Set up
+
+```{r}
+# This logic is the same as analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd
+# Assigning params$is_ci to running_in_ci avoids a locked binding error
+running_in_ci <- params$is_ci
+
+# Are we testing? In case of a non 0/1 number, we recast as logical, and then 
+# ensure logical.
+if (running_in_ci %in% c(0,1)) running_in_ci <- as.logical(running_in_ci)
+if (!(is.logical(running_in_ci)))
+{
+  stop("\n\nERROR: The parameter `is_ci` should be FALSE/TRUE (or 0/1).")
+}
+
+
+library(tidyverse)
+
+if (running_in_ci) {
+  data_dir <- "../../data/"
+} else {
+data_dir <- "../../data/release-v18-20201123/"
+}
```

Comment:
This conditional code should not be needed. The way the system is set up, all files should be linked to `data/` with no subfolder. In CI, this will be the testing files, but in regular use (after running `download-data.sh`) it will be the release version. Unless we want to make this file only work in v18 (I don't think we do) we should not be referring to the subfolder. 
```suggestion
data_dir <- "../../data/"
```

Reply:
hmm - @jashapiro I think we did this because it was being run before CI files were generated. However, also thinking about the process, it is a bit circular. This needs to be run for `pbta-histologies.tsv` to be generated and added to the new release, before the CI files are generated. Thoughts on that?

Reply:
The biggest concern I have is that somebody might rerun this script when generating v19+ and forget to update this line.  I don't know if this is pervasive through the subtyping files. 

(Of course, with the linking approach it is possible to have the wrong set of files linked, and get a hidden error that way, so neither one is great.)

Reply:
@kgaonkar6 @jashapiro what if we add this as a requirement in the bash script in which the user has to put a release version?

Reply:
I just looked at some of the other subtyping analyses, and we don't do that for those. I think for consistency with the other analyses, it should simply use `data/` and expect that the correct files have been placed or linked there.

Reply:
ok sounds good, I'll update the code to always use data